app/src/main/cpp/skyline/gpu/texture/layout.cpp

```diff
@@ -89,144 +89,428 @@ namespace skyline::gpu::texture {
 
         u8 *sector{blockLinear};
 
-        auto deswizzleRob{[&](u8 *linearRob, auto isLastRob, size_t blockPaddingY = 0, size_t blockExtentY = 0) {
-            auto deswizzleBlock{[&](u8 *linearBlock, auto copySector) __attribute__((always_inline)) {
+        auto deswizzleRob{[&](u8 *pitchRob, auto isLastRob, size_t blockPaddingY = 0, size_t blockExtentY = 0) {
+            auto deswizzleBlock{[&](u8 *pitchBlock, auto copySector) __attribute__((always_inline)) {
                 for (size_t gobZ{}; gobZ < blockDepth; gobZ++) { // Every Block contains `blockDepth` Z-axis GOBs (Slices)
-                    u8 *linearGob{linearBlock};
+                    u8 *pitchGob{pitchBlock};
                     for (size_t gobY{}; gobY < blockHeight; gobY++) { // Every Block contains `blockHeight` Y-axis GOBs
                         #pragma clang loop unroll_count(SectorLinesInGob)
                         for (size_t index{}; index < SectorLinesInGob; index++) {
                             size_t xT{((index << 3) & 0b10000) | ((index << 1) & 0b100000)}; // Morton-Swizzle on the X-axis
                             size_t yT{((index >> 1) & 0b110) | (index & 0b1)}; // Morton-Swizzle on the Y-axis
 
                             if constexpr (!isLastRob) {
-                                copySector(linearGob + (yT * robWidthUnalignedBytes) + xT, xT);
+                                copySector(pitchGob + (yT * pitchWidthBytes) + xT, xT);
                             } else {
                                 if (gobY != blockHeight - 1 || yT < blockExtentY)
-                                    copySector(linearGob + (yT * robWidthUnalignedBytes) + xT, xT);
+                                    copySector(pitchGob + (yT * pitchWidthBytes) + xT, xT);
                                 else
                                     sector += SectorWidth;
                             }
                         }
 
-                        linearGob += gobYOffset; // Increment the linear GOB to the next Y-axis GOB
+                        pitchGob += gobYOffset; // Increment the linear GOB to the next Y-axis GOB
                     }
 
-                    linearBlock += gobZOffset; // Increment the linear block to the next Z-axis GOB
+                    if constexpr (isLastRob)
+                        sector += blockPaddingY; // Skip over any padding at the end of this slice
+
+                    pitchBlock += gobZOffset; // Increment the linear block to the next Z-axis GOB
                 }
 
                 sector += blockPaddingZ; // Skip over any padding Z-axis GOBs
             }};
 
             for (size_t block{}; block < robWidthBlocks; block++) { // Every ROB contains `surfaceWidthBlocks` blocks (excl. padding block)
-                deswizzleBlock(linearRob, [&](u8 *linearSector, size_t) __attribute__((always_inline)) {
-                    if constexpr (BlockLinearToLinear)
-                        std::memcpy(linearSector, sector, SectorWidth);
```

Comment:
The compiler will optimise memcpy here anyway so manually doing it doesn't really benefit much